src/main/java/org/apache/sling/servlets/post/impl/operations/DeleteOperation.java

```diff
@@ -69,6 +71,9 @@ protected void doRun(final SlingHttpServletRequest request,
         final Iterator<Resource> res = getApplyToResources(request);
         if (res == null) {
             final Resource resource = request.getResource();
+            if (resource instanceof NonExistingResource) {
```

Comment:
I think this could still use some clarification about what should happen in the alternate code path where the resources to delete are provided by one or more ":applyTo" request parameters.  If I am not mistaken, this looks like you are only checking if the resource exists in the simple case where the resource to delete is itself and no ":applyTo" arguments are provided?